### PR TITLE
Correctly substring fields before update, remove duplicates and add missing sql queries

### LINKED ISSUE #21948: Problem with ALTER table during Update
#### Describe the bug

Problem with ALTER table during Update

#### Steps to Reproduce

Make an update 1769 --> 1770

#### Additional information

Linux #799710 SMP Wed Jul 10 08:31:28 UTC 2019 x86_64

Version du logiciel serveur : Apache

Version de PHP : 7.2.34
Limite de mémoire : 2048M
Temps maximal d'exécution : 18000
Taille max. pour envoi de fichiers : 10M
info_outline Informations sur la base de données

Version de MySQL : 10.1.45-MariaDB-0+deb9u1
Serveur MySQL : localhost
Nom MySQL : 176
Utilisateur MySQL : 176
Préfixe des tables : ps_
Moteur MySQL : InnoDB
Connecteur MySQL : DbPDO

```
SQL 1.7.7.0 1071 in ALTER TABLE `ps_admin_filter` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_alias` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_authorization_role` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_module_preference` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_smarty_lazy_cache` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_tab_module_preference` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_admin_filter` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_alias` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_authorization_role` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_module_preference` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_smarty_lazy_cache` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_tab_module_preference` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
```

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Remove duplicates. Make sure the fields are truncated before changing the size for the utf8mb4 support
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #21948 .
| How to test?  | Update from 1.7.6.x to 1.7.7.x.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21975)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

File: install-dev/upgrade/sql/1.7.7.0.sql [hunk 1]
```diff
@@ -22,11 +22,32 @@ DELETE FROM `PREFIX_configuration` WHERE `name` = 'PS_PRICE_DISPLAY_PRECISION';
...
 ALTER TABLE `PREFIX_hook_alias` CHANGE `name` `name` VARCHAR(191) NOT NULL;
 ALTER TABLE `PREFIX_hook_alias` CHANGE `alias` `alias` VARCHAR(191) NOT NULL;
 
+UPDATE `PREFIX_admin_filter` SET `filter_id` = SUBSTRING(`filter_id`, 1, 191);
+ALTER TABLE `PREFIX_admin_filter` CHANGE `filter_id` `filter_id` VARCHAR(191) NOT NULL;
+
+UPDATE `PREFIX_alias` SET `alias` = SUBSTRING(`alias`, 1, 191);
+ALTER TABLE `PREFIX_alias` CHANGE `alias` `alias` VARCHAR(191) NOT NULL;
+
+UPDATE `PREFIX_hook_authorization_role` SET `slug` = SUBSTRING(`slug`, 1, 191);
```
Comment (ghost) [inline install-dev/upgrade/sql/1.7.7.0.sql]: PREFIX_hook_authorization_role
PREFIX_authorization_role

File: install-dev/upgrade/sql/1.7.7.0.sql [hunk 2]
```diff
@@ -22,11 +22,32 @@ DELETE FROM `PREFIX_configuration` WHERE `name` = 'PS_PRICE_DISPLAY_PRECISION';
...
+ALTER TABLE `PREFIX_authorization_role` CHANGE `slug` `slug` VARCHAR(191) NOT NULL;
+
+UPDATE `PREFIX_module_preference` SET `module` = SUBSTRING(`module`, 1, 191);
+ALTER TABLE `PREFIX_module_preference` CHANGE `module` `module` VARCHAR(191) NOT NULL;
+
+UPDATE `PREFIX_hook_tab_module_preference` SET `module` = SUBSTRING(`module`, 1, 191);
+ALTER TABLE `PREFIX_hook_tab_module_preference` CHANGE `module` `module` VARCHAR(191) NOT NULL;
```
Comment (ghost) [inline install-dev/upgrade/sql/1.7.7.0.sql]: PREFIX_hook_tab_module_preference
PREFIX_tab_module_preference

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
It's ok for me.

File: install-dev/upgrade/sql/1.7.7.0.sql [hunk 3]
```diff
@@ -22,11 +22,32 @@ DELETE FROM `PREFIX_configuration` WHERE `name` = 'PS_PRICE_DISPLAY_PRECISION';
 ALTER TABLE `PREFIX_employee` MODIFY COLUMN `optin` tinyint(1) unsigned DEFAULT NULL;
 
 /* Increase column size */
+UPDATE `PREFIX_hook` SET `name` = SUBSTRING(`name`, 1, 191);
```
Comment (atomiix) [inline install-dev/upgrade/sql/1.7.7.0.sql:25]: Is it really needed here? It was `varchar(64)` before so it cannot be > 191.

File: install-dev/upgrade/sql/1.7.7.0.sql [hunk 4]
```diff
@@ -22,11 +22,32 @@ DELETE FROM `PREFIX_configuration` WHERE `name` = 'PS_PRICE_DISPLAY_PRECISION';
...
 ALTER TABLE `PREFIX_hook` CHANGE `name` `name` VARCHAR(191) NOT NULL;
 ALTER TABLE `PREFIX_hook` CHANGE `title` `title` VARCHAR(255) NOT NULL;
+
+UPDATE `PREFIX_hook_alias` SET `name` = SUBSTRING(`name`, 1, 191), `alias` = SUBSTRING(`alias`, 1, 191);
```
Comment (atomiix) [inline install-dev/upgrade/sql/1.7.7.0.sql:29]: Same here

File: install-dev/upgrade/sql/1.7.7.0.sql (hunk 3 above)
Comment (PierreRambaud) [reply install-dev/upgrade/sql/1.7.7.0.sql:25]: The size of the field was updated to 191 in another pull request.

**Review by Progi1984 — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Review by eternoendless — APPROVED** [review]
LGTM

**Comment by SD1982** [comment]
LGTM 
Tested with :
- php 7.2.33
- apache 2.4.38
- mysql 5.7.31 & mariadb 10.1.45 

No more error during the upgrade process :+1: 
PS : the initial error described in #21948 was only displayed with mariadb no error with mysql

Thanks @PierreRambaud !!

**Comment by Progi1984** [comment]
Thanks @SD1982 & @PierreRambaud

**Pull request merged**